docs/load_balancer.md

```diff
@@ -0,0 +1,278 @@
+# Load balancing & publicly accessible services
+
+This document describes how kubernetes services are made accessible outside the cluster.
+
+There are two main mechanisms:
+
+* The service can be exposed through a load-balancer managed by k8s.
+* k8s can also expose services without an associated load-balancer; the service could be reached via DNS SRV records,
+or by a load-balancer configured by a helper process, or even by a client using the k8s API directly for discovery.
+_(new functionality)_
+
+## Public ports
+
+To be exposed outside the k8s cluster, the service must be on known ports on externally reachable
+IPs.  For version 1 public services, k8s listens on _every_ external IP address on every minion, and assigns
+a _single_ port cluster-wide for each port on public services.
+
+## Important scenarios
+
+The names in bold are the names we will use in the API.
+
+1. **cluster** The service is reachable from within the cluster.  _(Existing functionality)_
+1. **loadbalancer** The service is exposed via a k8s-managed load-balancer.  _(Existing functionality)_
+1. **public** The service is exposed on a set of publicly-routable IP addresses, on a set of ports.  _(New functionality)_.
```

Comment:
Good point.  I was going to dial down the spec to "The service is exposed on a set of nodes, on a set of ports", and then say "The intention is that this service would then be reachable from outside the k8s cluster; a cloud-provider should automatically configure firewall rules (or equivalents) so that a public service is reachable from the public Internet."

But that prompts the obvious realization that some people will want public services that are reachable only from their other cloud instances.  This could be another visibility value.  Are we OK leaving this out for now?